src/main/java/frc/robot/commands/FieldOrientedWithCardinal.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import com.pathplanner.lib.util.PIDConstants;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.trajectory.TrapezoidProfile.Constraints;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.classes.Structs;
+import frc.robot.subsystems.chassis.Chassis;
+import frc.robot.subsystems.chassis.PoseEstimator;
+import java.util.function.DoubleSupplier;
+import java.util.function.Supplier;
+
+public class FieldOrientedWithCardinal extends Command {
+  private final Chassis chassis;
+  private final Supplier<ChassisSpeeds> speedsSupplier;
+  private final PoseEstimator poseEstimator;
+  private final DoubleSupplier direction;
+
+  private final ProfiledPIDController thetaPID;
+  private final SimpleMotorFeedforward thetaFF;
+  private ChassisSpeeds speeds;
+
+  /** Creates a new FieldOrientedDrive. */
+  public FieldOrientedWithCardinal(
+      Chassis chassis,
+      PoseEstimator poseEstimator,
+      DoubleSupplier direction,
```

Comment:
I don't think this needs to be a supplier. The double can be calculated when the command is invoked and passed in as a normal double

Reply:
I think that if we want to have the command run while the cardinal buttons are being changed, we need to have a supplier. I think it also makes it easier to run the direction calculation